blockchain/chain_sync/src/sync.rs

```diff
@@ -1103,9 +1103,12 @@ mod tests {
         task::block_on(async move {
             cs.peer_manager.add_peer(source.clone(), None).await;
             assert_eq!(cs.peer_manager.len().await, 1);
-
-            let return_set = cs.sync_headers_reverse(head, &to).await;
-            assert_eq!(return_set.unwrap().len(), 4);
+            // make blocksync request
+            let return_set = task::spawn(async move { cs.sync_headers_reverse(head, &to).await });
+            task::sleep(Duration::from_secs(2)).await;
```

Comment:
Is the reason for the thread sleep because the message published does not get queued and can only be read when polling? Just trying to understand the potential lost messages

Reply:
Yeah its just to ensure a request gets made before a response gets sent